src/lib.rs

```diff
@@ -732,6 +732,29 @@ pub trait Itertools : Iterator {
         adaptors::unique(self)
     }
 
+    /// Check whether all elements compare equal.
+    ///
+    /// Empty iterators are considered to have equal elements:
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![1, 1, 1, 2, 2, 3, 3, 3, 4, 5, 5];
+    /// assert!(!data.iter().all_equal());
+    /// assert!(data[0..3].iter().all_equal());
+    /// assert!(data[3..5].iter().all_equal());
+    /// assert!(data[5..8].iter().all_equal());
+    ///
+    /// let data : Option<usize> = None;
+    /// assert!(data.into_iter().all_equal());
+    /// ```
+    fn all_equal(self) -> bool 
```

Comment:
By the way, the method should be sorted among the methods, not the adaptors, so it needs to be moved down. next_tuple is the first non-adaptor in the trait.